src/Symfony/Bridge/Twig/Extension/TranslationExtension.php

```diff
@@ -34,7 +34,7 @@ final class TranslationExtension extends AbstractExtension
     private $translator;
     private $translationNodeVisitor;
 
-    public function __construct(TranslatorInterface $translator = null, NodeVisitorInterface $translationNodeVisitor = null)
+    public function __construct(TranslatorInterface $translator = null, TranslationNodeVisitorInterface $translationNodeVisitor = null)
```

Comment:
Let's not create a new interface and type hint with `TranslationNodeVisitor` instead.

Reply:
That would limit flexibility without much benefit.

I have a use case for passing a different node visitor into this class (basically overwriting the trans function and adding an extra parameter). That was possible in symfony 4.4, because there was no return typehint on the getter. I came across this trying to upgrade to symfony ^5.0.

It's up to you. If you still prefer the concrete class typehint then I will change it. Let me know @fabpot 👍